### PR TITLE
Add .editorconfig and cleanup syntax

### PR DESCRIPTION
**Pull request by nicolas-brousse**
This PR does the following:

- Adding an `.editorconfig` file to try having all the same syntax on the project
- Enforce to use **single quotes** on the code (default on `rubocop`)
- Use **double quotes** on all **javascripts** generated files (default on `eslint`)

### CODE REVIEW

File: lib/generators/komponent/styleguide_generator.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
```
Comment (florentferry) [inline lib/generators/komponent/styleguide_generator.rb]: That file seems not belong to this PR.

**Comment by Spone** [comment]
> Enforce to use single quotes on the code (default on rubocop)

I know it's the default on Rubocop but personally, I prefer to use double quotes. What do you think @florentferry?

File: lib/generators/komponent/styleguide_generator.rb (hunk 1 above)
Comment (nicolas-brousse) [reply lib/generators/komponent/styleguide_generator.rb]: Hmmm looks like.

**Comment by nicolas-brousse** [comment]
I also prefer double, but since most of ruby dev seems to use single in OSS, I used single here to start. Just let me know your point of vue.

**Comment by florentferry** [comment]
Ok for single quote.

**Comment by Spone** [comment]
Rails uses double quotes btw: https://github.com/rubocop-hq/rubocop/issues/5306#issuecomment-354273615

File: lib/generators/komponent/styleguide_generator.rb (hunk 1 above)
Comment (florentferry) [reply lib/generators/komponent/styleguide_generator.rb]: Can you remove it ?

Comment (nicolas-brousse) [reply lib/generators/komponent/styleguide_generator.rb]: Yes I will, sorry I was really busy those days

Comment (nicolas-brousse) [reply lib/generators/komponent/styleguide_generator.rb]: Done

**Comment by Spone** [comment]
@nicolas-brousse after discussing it over lunch with @florentferry: we think we can stick with single-quotes, but it would be great to trigger `rubocop --auto-correct` on the generated component directory, so the partial is corrected with the project preferred style. What do you think?

**Comment by nicolas-brousse** [comment]
* What about javascript and css? If we do this fix for ruby, why not for other files?
* What if rubocop is not installed on project?
* If we run this command, it mustn't be blocking at all and avoid to brake generation if something wrong with rubocop config for example

**Comment by nicolas-brousse** [comment]
Also it's not a behavior I already so in other generation system. May be it is for a good reason? (real question)

**Comment by florentferry** [comment]
Rails generates project with double quotes, and generates scaffold with double quotes. If you install Rubocop with default config on top of that, you need to change several files from double quotes to single quote. Use rubocop with `-a` option do the trick.

**Comment by Spone** [comment]
> Also it's not a behavior I already so in other generation system. May be it is for a good reason? (real question)

It's a good question. I'm doing some research, I'll let you know if I find something.

**Comment by Spone** [comment]
I did not find any other library doing that.

Maybe we could add it as an option?

```ruby
config.generators do |g|
  g.komponent rubocop: true # this would be false by default
end
```

In this case `rubocop -a` would be run after the generator.

What do you think?

**Comment by Spone** [comment]
I'm closing this stale PR, there are too many conflicts now. Feel free to reopen another one.
